src/metpy/plots/declarative.py

```diff
@@ -1831,3 +1835,266 @@ def _build(self):
+
+        # If object is a MultiPolygon or MultiLineString, associate the label with the single
+        # largest Polygon or LineString from the collection. If MultiPoint, associate the label
+        # with one of the Points in the MultiPoint, chosen based on the label hash.
+        if isinstance(geo_obj, (MultiPolygon, MultiLineString)):
+            geo_obj = max(geo_obj, key=lambda x: x.length)
+        elif isinstance(geo_obj, MultiPoint):
+            geo_obj = geo_obj[label_hash % len(geo_obj)]
+
+        # Get the list of coordinates of the polygon/line/point
+        if isinstance(geo_obj, Polygon):
+            coords = geo_obj.exterior.coords
+        else:
+            coords = geo_obj.coords
+
+        position = coords[label_hash % len(coords)]
+
+        return position
+
+    def draw_label(self, text, lon, lat, color='black', outline='white', offset=(0, 0)):
+        """Draw a label to the plot.
+
+        Parameters
+        ----------
+        text : str
+            The label's text
+        lon : float
+            Longitude at which to position the label
+        lat : float
+            Latitude at which to position the label
+        color : str (default: 'black')
+            Name or hex code for the color of the text
+        outline : str (default: 'white')
+            Name or hex code of the color of the outline of the text
+        offset : tuple (default: (0, 0))
+            A tuple containing the x- and y-offset of the label, respectively
+        """
+        path_effects = [patheffects.withStroke(linewidth=4, foreground=outline)]
+        self.parent.ax.add_collection(TextCollection([lon], [lat], [str(text)],
+                                                     va='center',
+                                                     ha='center',
+                                                     offset=offset,
+                                                     weight='demi',
+                                                     size=self.label_fontsize,
+                                                     color=color,
+                                                     path_effects=path_effects,
+                                                     transform=ccrs.PlateCarree()))
+
+    def draw(self):
+        """Draw the plot."""
+        if self._need_redraw:
+            if getattr(self, 'handles', None) is None:
+                self._build()
+            self._need_redraw = False
+
+    def copy(self):
+        """Return a copy of the plot."""
+        return copy.copy(self)
+
+    def _build(self):
+        """Build the plot by calling needed plotting methods as necessary."""
+        # Each Shapely object is plotted separately
+        for i, geo_obj in enumerate(self.geometry):
+            # Handle `fill` and `stroke` when they are None
+            self.stroke = ['none'] if self.stroke is None else self.stroke
+            self.fill = ['none'] if self.fill is None else self.fill
```

Comment:
Could these be handled in validators? Also, I'm not sure we're handling the case where a single fill/stroke is passed in as a string? Unless I'm missing something, it's going to be getting a single character?

Reply:
I'd really like to handle these in the `_valid_color_list()` validator, but traitlets doesn't use the validator when the proposed value is `None` (sigh). For that reason, I had been going back-and-forth a lot about whether to even use validators. If you have a better way to structure all of this "making sure colors are lists" stuff, I'm all ears.

Also, it looks like are currently handling the single string case you mentioned in `_valid_color_list()`.